src/widgets/details/factory.rs

```diff
@@ -357,13 +383,22 @@ impl AsyncFactoryComponent for TaskDetailsFactoryModel {
 				adw::PreferencesGroup::default(),
 				sender.input_sender(),
 			),
+			tags: FactoryVecDeque::new(
+				gtk::FlowBox::default(),
+				sender.input_sender(),
+			),
 			dirty: false,
 		};
-		let mut guard = model.sub_tasks.guard();
+		let mut sub_tasks_guard = model.sub_tasks.guard();
 		for sub_task in init.task.sub_tasks {
-			guard.push_back(SubTaskInit::new(sub_task));
+			sub_tasks_guard.push_back(SubTaskInit::new(sub_task));
+		}
+		let mut tags_guard = model.tags.guard();
+		for tag in init.task.tags {
+			tags_guard.push_back(TagInit::new(tag));
 		}
-		guard.drop();
+		sub_tasks_guard.drop();
+		tags_guard.drop();
```

Comment:
You usually don't need to drop the guard manually.

Reply:
Here's the error message I get when I remove those two lines:

```fish
error[E0505]: cannot move out of `model` because it is borrowed
   --> src/widgets/details/factory.rs:400:3
    |
392 |         let mut sub_tasks_guard = model.sub_tasks.guard();
    |                                   ----------------------- borrow of `model.sub_tasks` occurs here
...
400 |         model
    |         ^^^^^ move out of `model` occurs here
401 |     }
    |     - borrow might be used here, when `sub_tasks_guard` is dropped and runs the `Drop` code for type `FactoryVecDequeGuard`
```

Reply:
I see. Probably using a scope would be the best solution then.

Reply:
That works 👍🏼